src/test/scala/aserralle/akka/stream/kcl/KinesisWorkerSourceSourceSpec.scala

```diff
@@ -113,10 +115,61 @@ class KinesisWorkerSourceSourceSpec
         verify(worker).run()
       }
     }
+
+    "not drop messages in case of back-pressure" in new KinesisWorkerContext
+    with TestData {
+      recordProcessor.initialize(initializationInput)
+      for (i <- 1 to 10) { // 10 is a buffer size
+        val record = org.mockito.Mockito.mock(classOf[Record])
+        when(record.getSequenceNumber).thenReturn(i.toString)
+        recordProcessor.processRecords(
+          recordsInput.withRecords(List(record).asJava))
+      }
+      //expect to consume all 10
+      for (_ <- 1 to 10) sinkProbe.requestNext()
+
+      //send another batch to exceed the queue size
+      Future {
+        for (i <- 1 to 25) { // 10 is a buffer size
+          val record = org.mockito.Mockito.mock(classOf[Record])
+          when(record.getSequenceNumber).thenReturn(i.toString)
+          recordProcessor.processRecords(
+            recordsInput.withRecords(List(record).asJava))
+        }
+      }
+
+      //expect to consume all 25 with slow consumer
+      for (_ <- 1 to 25) {
+        sinkProbe.requestNext()
+        Thread.sleep(100)
+      }
+
+      killSwitch.shutdown()
+      sinkProbe.expectComplete()
+    }
+
+    "stop the stream when back pressure timeout elapsed" in new KinesisWorkerContext(
+      backpressureTimeout = 100.milliseconds) with TestData {
+      recordProcessor.initialize(initializationInput)
+      //Fast consumer sends 25 messages into 10 items queue size
+      for (i <- 1 to 25) { // 10 is a buffer size
+        val record = org.mockito.Mockito.mock(classOf[Record])
+        when(record.getSequenceNumber).thenReturn(i.toString)
+        recordProcessor.processRecords(
+          recordsInput.withRecords(List(record).asJava))
+      }
+
+      Await.ready(watch, 5.seconds)
+      val Failure(exception) = watch.value.get
+      assert(exception.getMessage == "Back-pressure exhausted")
```

Comment:
How is this assertion still passing? 😕

Reply:
Yes this doesn't make sense, let me check

Reply:
i don't see any tests are actually running on Travis. can you take a look ?